### PR TITLE
Add e2e test for iOS/Android

### PR DESCRIPTION
**Pull request by denrase**
#skip-changelog 

## :scroll: Description

- Adds an e2e test to iOS & Android integration tests
- Opted to extend the example app after all, but with a section that is only visible when the integration test is run.
- That way we don't need to keep two apps up to date and we use the same options

## :bulb: Motivation and Context

Relates to #514

## :green_heart: How did you test it?

Test

## :pencil: Checklist

- [x] I reviewed submitted code
- [x] I added tests to verify changes
- [ ] No new PII added or SDK only sends newly added PII if `sendDefaultPii` is enabled
- [ ] I updated the docs if needed
- [x] All tests passing
- [x] No breaking changes

### CODE REVIEW

**Comment by marandaneto** [comment]
if we go down this way, let's migrate https://github.com/getsentry/sentry-dart/blob/main/flutter/example/integration_test/integration_test.dart to this example so we have a single place to test stuff.

**Comment by krystofwoldrich** [comment]
Would there be any benefit of running E2E tests with multiple Flutter versions like we have it in RN?

**Comment by marandaneto** [comment]
> Would there be any benefit of running E2E tests with multiple Flutter versions like we have it in RN?

Yes, but the overhead is quite big (CI time/maintenance), we run in the minimal supported version, the stable and beta version, which is IMO enough.

File: flutter/example/lib/main.dart [hunk 1]
```diff
@@ -63,6 +66,12 @@ Future<void> setupSentry(AppRunner appRunner, String dsn) async {
 
     options.maxRequestBodySize = MaxRequestBodySize.always;
     options.maxResponseBodySize = MaxResponseBodySize.always;
+
+    _isIntegrationTest = isIntegrationTest;
+    if (_isIntegrationTest) {
+      options.dist = '1';
+      options.environment = 'integration';
```
Comment (marandaneto) [inline flutter/example/lib/main.dart:73]: Wondering why we need the `dist`?

File: flutter/example/pubspec.yaml [hunk 2]
```diff
@@ -27,6 +27,7 @@ dependencies:
   path_provider: ^2.0.0
   #sqflite_common_ffi: ^2.0.0
   #sqflite_common_ffi_web: ^0.3.0
+  http: ^0.13.0
```
Comment (marandaneto) [inline flutter/example/pubspec.yaml]:
```suggestion
  http: ^1.0.0
```

File: flutter/example/integration_test/integration_test.dart [hunk 3]
```diff
@@ -124,4 +132,77 @@ void main() {
...
+      } finally {
+        retries++;
+        await Future.delayed(delay);
+        delay *= factor;
+      }
+    }
+    return null;
+  }
+
+  bool validate(Map<String, dynamic> event) {
```
Comment (marandaneto) [inline flutter/example/integration_test/integration_test.dart]: Can we actually validate more properties in the event?
I know this is a copy of https://github.com/getsentry/sentry-dart/blob/main/e2e_test/bin/e2e_test.dart
Maybe we should improve that test as well.
The goal is that we assert every property that is static and not dependent on the host platform, so it wont fail in different devices, etc.

**Comment by krystofwoldrich** [comment]
> minimal supported version, the stable and beta version

Yes, I agree, this is enough for sure, I've missed that, I thought the e2e run only on stable.

File: flutter/example/integration_test/integration_test.dart (hunk 3 above)
Comment (denrase) [reply flutter/example/integration_test/integration_test.dart]: I check fro the same properties than https://github.com/getsentry/sentry-kotlin-multiplatform/pull/78 now. Do we need more? If not, i will update the dart only test as well.

Comment (marandaneto) [reply flutter/example/integration_test/integration_test.dart]: Let's start with the current approach and expand as we go.
Ideally, we'd check every property that is static, does not change depending on the device, etc.

File: flutter/example/android/app/src/main/AndroidManifest.xml [hunk 4]
```diff
@@ -1,6 +1,9 @@
 <manifest xmlns:android="http://schemas.android.com/apk/res/android"
     package="io.sentry.samples.flutter">
 
+    <uses-permission android:name="android.permission.INTERNET" />
```
Comment (marandaneto) [inline flutter/example/android/app/src/main/AndroidManifest.xml]: Android SDK already adds `INTERNET`.

**Review by marandaneto — APPROVED** [review]

**Pull request merged**